File: Java/DesignPatterns/src/main/java/designPatterns/structural/bridge/ConcreteImplementor.java
package designPatterns.structural.bridge;

public class ConcreteImplementor {
  static class A implements Implementor {
    @Override
    public String applyImplement() {
      return "Applying A";
    }
  }

  static class B implements Implementor {
    @Override
    public String applyImplement() {
      return "Applying B";
    }
  }
}

File: Kotlin/DesignPatterns/src/main/kotlin/structural/bridge/ConcreteImplementor.kt
package structural.bridge

sealed class ConcreteImplementor {
  class A : Implementor {
    override fun applyImplementor(): String {
      return "Applying A"
    }
  }

  class B : Implementor {
    override fun applyImplementor(): String {
      return "Applying B"
    }
  }
}
